packages/astro/src/core/app/index.ts

```diff
@@ -280,7 +284,7 @@ export class App {
 			}
 
 			// Validate X-Forwarded-Host against allowedDomains if configured
-			if (forwardedHost && !this.matchesAllowedDomains(forwardedHost, protocol)) {
+			if (forwardedHost && !this.matchesAllowedDomains(forwardedHost, protocol?.replace(':', ''))) {
```

Comment:
What's this replace?

Reply:
It's when the protocol includes a `:` actually, I just noticed the code above, so `https:` -> `https`, got it